classes/class06/twoter/app.js

```diff
@@ -33,9 +34,17 @@ app.use(session({
   resave: false,
   saveUninitialized: true
 }));
-// app.use(passport.initialize());
-// app.use(passport.session());
+app.use(passport.initialize());
+app.use(passport.session());
 app.use(express.static(path.join(__dirname, 'public')));
 
+// Routes.
 require('./routes');
+
+// Test authentication.
+function ensureAuthenticated(req, res, next) {
```

Comment:
You define this function but never call it. That's not a bad thing though, in this case, because it redirects to `'/'` if the user isn't logged in, which will just redirect them to `'/login'`. You want to minimize the amount of redirecting you have to do.